src/ServiceInsight/ServiceControl/DefaultServiceControl.cs

```diff
@@ -161,7 +161,25 @@ public void LoadBody(StoredMessage message)
             {
                 baseUrl = null; // We use the default
             }
-            message.Body = Execute(request, response => CleanupBodyString(response.Content), baseUrl);
+
+            message.Body = Execute(request, response =>
+            {
+                var presentationBody = new PresentationBody();
+
+                switch (response.StatusCode)
+                {
+                    case HttpStatusCode.OK:
+                        presentationBody.Text = CleanupBodyString(response.Content);
+                        break;
+                    case HttpStatusCode.NoContent:
+                        presentationBody.Hint = PresentationHint.NoContent;
+                        presentationBody.Text = "Body was too large and not stored. Edit ServiceControl/MaxBodySizeToStore to be larger in the ServiceControl configuration.";
```

Comment:
No only put it there to assist debugging later on, if you were maintaining this a year from now and were trying to work out what the NoContent means this little snippet would be pretty useful, also would appear anywhere we just use text instead of special XAML views.